src/subject_name/dns_name.rs

```diff
@@ -150,6 +150,26 @@ pub enum GeneralDnsNameRef<'name> {
     Wildcard(WildcardDnsNameRef<'name>),
 }
 
+impl<'name> GeneralDnsNameRef<'name> {
+    /// Constructs a [DnsNameRef] from the given input if the input is a
+    /// syntactically-valid DNS name, or a [WildcardDnsNameRef] if the input is a
+    /// syntactically-valid DNS name with a wildcard label.
+    pub fn try_from_ascii(dns_name: &'name [u8]) -> Result<Self, InvalidDnsNameError> {
+        DnsNameRef::try_from_ascii(dns_name)
```

Comment:
Style nit: I know this is copy-pasting, but I tend to dislike the `combinator().combinator()` style if the combinators work on "different sides" of the value. I feel like in these cases an explicit `match` will be easier to follow.